components/schema-migrator/migrations/director/20220301143225_ord_increase_short_description_length.down.sql

```diff
@@ -0,0 +1,526 @@
+BEGIN;
+
+-- all views below should be dropped because they depend on the 'short_description' column that will be later modified in the respective tables
+DROP VIEW tenants_packages;
+DROP VIEW tenants_products;
+DROP VIEW tenants_bundles;
+DROP VIEW tenants_specifications;
+DROP VIEW tenants_apis;
+DROP VIEW tenants_events;
+
+DROP VIEW packages_tenants;
+DROP VIEW products_tenants;
+DROP VIEW bundles_tenants;
+DROP VIEW api_definitions_tenants;
+DROP VIEW event_api_definitions_tenants;
+
+ALTER TABLE packages
+ALTER COLUMN short_description TYPE VARCHAR(255);
+
+ALTER TABLE bundles
+ALTER COLUMN short_description TYPE VARCHAR(255);
+
+ALTER TABLE products
+ALTER COLUMN short_description TYPE VARCHAR(255);
+
+ALTER TABLE api_definitions
+ALTER COLUMN short_description TYPE VARCHAR(255);
+
+ALTER TABLE event_api_definitions
+ALTER COLUMN short_description TYPE VARCHAR(255);
+
+
+-- while re-creating this view, the 'support_info' column had to be explicitly omitted (thus selecting the rest columns one by one). This is needed because when it was introduced in
+-- "add-support-info-to-packages.up" migration, this view was not adapted to include it. In the current "up" migration, the 'support_info' column is included
+-- because all package columns are selected there (p.*). If we do not exclude the "support_info" column here, the "add-support-info-to-packages.down"
+-- migration will fail because when trying to drop the column this view will already rely on it.
```

Comment:
Does that mean that if we have a `p.*` in the view and we add a new column to the underlying table then the view is not automatically updated to have this column and need to be plain recreated in order to include the new column?

If that is the case and the reason for this headaches then we need to think of a way of simplifying contribution of migrations as this means that we need to recreate each and every view when adding new columns.

Reply:
To your question: yes, that is the case. In case a new column is added, these ("storage-redesign") views are not automatically updated to include the newly added column. And if in the future, that view is recreated in other migration , then it will include the new column (as we will select all table columns) which will break the initial down migration as it does not expect that view to depend on the new column.

So, for the time being, whenever we introduce a new column, we should make sure that all views that rely on that table are adapted to include it as well. However, as you suggested, in the long run it might be a better idea to come up with some unified solution that will solve this "error/forgetting prone" approach.